src/starterbot/ProductionManager.cpp

```diff
@@ -60,6 +60,29 @@ void ProductionManager::onFrame() {
     // Release any workers that are idle, i.e. not constructing any buildings.
     m_unitManager.releaseUnits(m_workers, bw::Filter::IsIdle);
 
+    // Check to see if refinery exists
+    for (bw::Unit unit : m_unitManager.borrowUnits(bw::Filter::IsRefinery)) {
```

Comment:
Why do you loop over every refinery? `m_gas_gatherers` will never have more than two units at any time. Did you mean to have two units per refinery, or just two units total?

If you just need two units total, then you can do
```cpp
if (m_unitManager.borrowUnit(bw::Filter::IsRefinery) != nullptr)
```
rather than a for loop. If you need two units per refinery, then you should probably do
```cpp
int gasCount = m_unitManager.borrowCount(bw::Filter::IsRefinery) * 2;
if (m_gas_gatherers.size() < gasCount) {
    bw::Unitset newGatherers = m_unitManager.reserveUnits(bw::Filter::IsWorker, gasCount - m_gas_gatherers.size());
    m_gas_gatherers.insert(newGatherers.begin(), newGatherers.end());
}
```